api/src/main/java/com/community/api/endpoint/avisoft/custom/CustomCategoryController.java

```diff
@@ -0,0 +1,499 @@
+package com.community.api.endpoint.avisoft.custom;
+
+import com.broadleafcommerce.rest.api.endpoint.catalog.CatalogEndpoint;
+import com.broadleafcommerce.rest.api.exception.BroadleafWebServicesException;
+import com.broadleafcommerce.rest.api.wrapper.CategoriesWrapper;
+import com.broadleafcommerce.rest.api.wrapper.CategoryAttributeWrapper;
+import com.broadleafcommerce.rest.api.wrapper.CategoryWrapper;
+import com.community.api.services.exception.ExceptionHandlingService;
+import org.broadleafcommerce.core.catalog.domain.*;
+import org.broadleafcommerce.core.catalog.service.CatalogService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.web.bind.annotation.*;
+
+import javax.annotation.Resource;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/categoryCustom")
+public class CustomCategoryController extends CatalogEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(com.community.api.endpoint.avisoft.controller.Category.CustomCategoryEndpoint.class);
+    private static final String CATEGORYNOTFOUND = "Category not Found";
+    private static final String CATEGORYTITLENOTGIVEN = "Category MetaTitle not Given";
+    private static final String CATEGORYDESCNOTGIVEN = "Category Description not Given";
+
+    @Autowired
+    private ExceptionHandlingService exceptionHandlingService;
+
+    @Autowired
+    private CustomCategoryService customCategoryService;
+
+    @PersistenceContext
+    private EntityManager em;
+
+
+    @PostMapping("/add")
+    public ResponseEntity<?> addCategory(HttpServletRequest request, @RequestBody CategoryImpl categoryImpl) {
+        try {
+            if (catalogService == null) {
+                throw BroadleafWebServicesException.build(404).addMessage("Catalog service is not initialized.");
```

Comment:
Remove all  404, 500, etc.
 It's better to return like this

                return new ResponseEntity<>("customerService is null", HttpStatus.INTERNAL_SERVER_ERROR);

Reply:
updated